networksimulator.py

```diff
@@ -0,0 +1,128 @@
+import heapq
+import json
+import sys
+
+class NetworkSimulator(object):
+    """The main class for the network simulator.
+
+    The NetworkSimulator holds a reference to all elements of the network in
+    the form of dictionaries. Also, it has a priority queue of events based
+    on execution time. Finally, it holds a reference to a DataMetrics object
+    which contains all the data obtained by the simulator.
+
+    Attributes:
+        flows (dict): all flows with their id's as the keys
+        links (dict): all links with their id's as the keys
+        routers (dict): all routers with their id's as the keys
+        hosts (dict): all hosts with their id's as the keys
+        pq (arr): a heapq priority queue whose elements are a tuple in the
+            form (execution_time, event_id, f) where f is the function to be
+            run
+        data (DataMetrics): container of all data generated by the simulator
+        cur_time (float): a current time counter in milliseconds
+        num_active_flows (int): Number of currently active flows
+        event_counter (int): an event counter used to uniquely identify events
+
+    """
+
+    def __init__(self):
+        self.flows = {}
+        self.links = {}
+        self.routers = {}
+        self.hosts = {}
+
+        self.pq = []
+        # self.data = DataMetrics()
+
+        self._cur_time = 0
+        self._num_active_flows = 0
+        self._event_counter = 0
+
+    @property
+    def cur_time(self):
+        return self._cur_time
+
+    @cur_time.setter
+    def cur_time(self, time):
+        self._cur_time = time
+
+    @property
+    def event_counter(self):
+        return self._event_counter
+
+    @event_counter.setter
+    def event_counter(self, val):
+        self._event_counter = val
+
+    def add_flow(self):
+        """Adds a new flow to the network.
+
+        num_active_flows is initialized to the number of flows.
```

Comment:
Num_active_flows isn't initialized in this function right? you just increment it